### DISCUSSION TITLE
SORTYBY Column REMOVEFILTERS bug not picked up

### DISCUSSION THREAD

**Discussion by wynhopkins** [TabularEditor/TabularEditor3#504] [Other Topics]
I was thinking a nice demo of DAX Debugger would be to demonstrate the bug where REMOVEFILTERS doesn't clear filters on a column such as month if you also use SORTBY Column on that column.

But Tabular Editor doesn't have the bug so the numbers display differently  (see image).   

Measure is 
`Full Year Budget = CALCULATE( [Budget], REMOVEFILTERS('Calendar'[Month] )  )`

Measure should be
`Full Year Budget = CALCULATE( [Budget], REMOVEFILTERS( 'Calendar'[Month], 'Calendar'[Financial Month Number for Sort] ) )`

![image](https://user-images.githubusercontent.com/44314473/162569798-7e3453ec-26ee-4499-af05-5a97c0d547c6.png)

**Comment by otykier** [comment]
Well, the Pivot Grid in TE3 uses MDX to query the model, and in that case the filter behaves differently - that’s just one of the many fun quirks of DAX. To demonstrate the filter issue, you could use a DAX query captured from Power BI with the performance analyzer, instead of a Pivot Grid.

Note that we are also considering an optional “DAX mode” for the Pivot Grid, but we can’t provide a timeline currently.

**Reply by wynhopkins** [reply]
Ah yeah, which is why Excel doesn't have the bug.    What am I doing wrong with the paste and run here?  
![image](https://user-images.githubusercontent.com/44314473/162574423-cfb437ff-3e77-4b7f-88c2-4dd4fba25828.png)

**Reply by otykier** [reply]
You’re mistaking DAX *script* for a DAX *query*. These are two different concepts in TE3. You need to use the latter.

**Reply by wynhopkins** [reply]
Ah… thanks for the help
